### DISCUSSION TITLE
Use alias to query parameter without setting the default value?

### DISCUSSION THREAD

**Discussion by zdytch** [fastapi/fastapi#6681] [Questions] [answered]
Let's say, my API endpoint must implement a required parameter 'from' by contract. I cannot use 'from' directly, as it's one of the Python reserved words. So I use the alias feature:

```
@router.get('/test/')
async def get_test(from_: int = Query(0, alias='from')):
	pass
```

Now I can use 'from' parameter, but it becomes optional, as Query object requires the 'default' argument to be set. Any ideas how to get a query parameter aliased and required at the same time?

**Comment by Kludex** [comment, marked as answer]
```python
@router.get('/test/')
async def get_test(from_: int = Query(..., alias='from')):
	pass
```

**Comment by zdytch** [comment]
@Kludex great, thank you!!
